packages/coverage-istanbul/src/provider.ts

```diff
@@ -175,6 +183,44 @@ export class IstanbulCoverageProvider implements CoverageProvider {
       }
     }
   }
+
+  async includeUntestedFiles(coverageMap: CoverageMap) {
+    // Load, instrument and collect empty coverages from all files which
+    // are not already in the coverage map
+    const includedFiles = await this.testExclude.glob(this.ctx.config.root)
+    const uncoveredFiles = includedFiles
+      .map(file => resolve(this.ctx.config.root, file))
+      .filter(file => !coverageMap.data[file])
+
+    const maxThreads = this.ctx.config.maxThreads ?? Math.max(cpus().length - 1, 1)
```

Comment:
I feel like this is bad idea.

1. `maxThreads` is not meant to be used like this
2. `cpus()`just returns virtual cores, which are usually not a lot

As I said before, we can be very liberal with this values - you proposed 25-30, I was thinking much more (transforming is not a hard operation), but in this PR we might even get 1 for `--no-threads`.

This value can be much higher, we don't even have limits in other places.

Reply:
@antfu can you back me up on this? is this premature optimisation?

Reply:
Alright, I must have misunderstood this part.

> maybe base it on **cores**/available memory?

I've now set the limit to a randomly picked value of 50. If this indeed seems unnecessary optimization, I'm open to remove it completely and run all transforms in a single `Promise.all`.

Reply:
I guess we could use `Promise.all` here, it shouldn't have much difference at this operation.

Reply:
Now using `Promise.all` for all transforms, without `p-limit`.